l2geth/rollup/sync_service.go

```diff
@@ -756,9 +760,11 @@ func (s *SyncService) verifyFee(tx *types.Transaction) error {
 	if !fee.IsUint64() {
 		return fmt.Errorf("fee overflow: %s", fee.String())
 	}
+	paying := new(big.Int).Mul(new(big.Int).SetUint64(tx.Gas()), tx.GasPrice())
```

Comment:
Is this correct? `tx.Gas()` already contains `l2GasLimit`. Should we not be masking out the `l2GasLimit` from `gas`?